sources/packages/api/src/services/atbc/atbc.service.ts

```diff
@@ -0,0 +1,180 @@
+} from "../auth/token-cache.service.models";
+import {
+  ATBCIntegrationConfig,
+  ATBCCreateClientResponse,
+  ATBCCreateClientPayload,
+  ATBCPDCheckerResponse,
+  ATBCPDCheckerPayload,
+} from "../../types";
+import { Student } from "../../database/entities";
+import { StudentService } from "../../services";
+
+@Injectable()
+export class ATBCService {
+  constructor(
+    private readonly configService: ConfigService,
+    private readonly studentService: StudentService,
+  ) {
+    this.logger.log("[Created]");
+  }
+  /**
+   * Allow a singleton access to the ATBCService for areas
+   * of the application that are not part of the dependency
+   * injection framework.
+   */
+  public static get shared(): ATBCService {
+    return ATBCService.shared;
+  }
+
+  get config(): ATBCIntegrationConfig {
+    return this.configService.getConfig().ATBCIntegration;
+  }
+  /**
+   * Executes the token request
+   * @returns bearer header.
+   */
+  private async getConfig(): Promise<ATBCHeader> {
+    const token = await this.getAuthToken();
+    return {
+      headers: { Authorization: `Bearer ${token}` },
+      // (NOTE: this will disable client verification)
+      // TODO: add certificate for PROD
+      httpsAgent: new (require("https").Agent)({
+        rejectUnauthorized: false,
+      }),
+    };
+  }
+
+  /**
+   * Gets the authentication token value to authorize the ATBC Endpoints.
+   * @returns the token that is needed to bearer authentication.
+   */
+  private async getAuthToken(): Promise<string> {
+    const authResponse = await this.ATBCLogin();
+    return authResponse?.accessToken;
+  }
+
+  /**
+   * Executes the authentication on ATBC.
+   * @returns the result of a success full authentication or throws an exception
+   * in case the result is anything different from HTTP 200 code.
+   */
+  public async ATBCLogin(): Promise<ATBCAuthTokenResponse> {
+    try {
+      const agent = new (require("https").Agent)({
+        rejectUnauthorized: false,
+      });
+
+      const authRequest = await axios.post(
+        this.config.ATBCloginEndpoint,
+        {
+          usr: this.config.ATBCUserName,
+          pwd: this.config.ATBCPassword,
+          app: this.config.ATBCApp,
+        },
+        { httpsAgent: agent },
+      );
+      return authRequest?.data as ATBCAuthTokenResponse;
+    } catch (excp) {
+      this.logger.error(`Received exception while ATBC Authentication`);
+      this.logger.error(`${JSON.stringify(excp.response)} `);
+      throw excp;
+    }
+  }
+
+  /**
+   * Create client on ATBC.
+   * @returns the result of a success full authentication or throws an exception
+   * in case the result is anything different from HTTP 200 code.
+   */
+  public async ATBCCreateClient(
+    payload: ATBCCreateClientPayload,
+  ): Promise<ATBCCreateClientResponse> {
+    try {
+      const config = await this.getConfig();
+      const apiEndpoint =
+        this.config.ATBCEndpoint.slice(-1) === "/"
```

Comment:
I would recommend do not add more complexity to check the "/" every time. It will not be good repeating this logic in all places needed. Please consider that the root endpoint will not have the "/" or resolve this logic on config service to be centralized.